decoder.cpp

```diff
@@ -190,6 +198,108 @@ bool Decoder::readImageResources()
   return (length == 0);
 }
 
+double Decoder::getTypeNumber(const std::unique_ptr<OSType>& v)
+{
+  if (v->type() == OSTypeKey::LargeInteger) {
+    return static_cast<const OSTypeLargeInt*>(v.get())->value;
+  }
+  else if (v->type() == OSTypeKey::Long) {
+    return static_cast<const OSTypeInt*>(v.get())->value;
+  }
+  else if (v->type() == OSTypeKey::Double) {
+    return static_cast<const OSTypeDouble*>(v.get())->value;
+  }
+  throw std::runtime_error("wrong descriptor type");
+};
+
+// the parsing used here is written thanks to the description given here
+// https://community.adobe.com/t5/photoshop-ecosystem-discussions/how-to-get-animation-frame-id/m-p/8790051#M68134
+std::unique_ptr<OSTypeDescriptor> Decoder::readAnimatedDataSection()
+{
+  read32(); // data length
+
+  const uint32_t descVersion = read32();
+  if (descVersion != 16)
+    return nullptr;
+
+  auto desc = parseDescriptor();
+  if (!desc)
+    return nullptr;
+
+  using DescriptorMap = OSTypeDescriptor::DescriptorMap;
+  const DescriptorMap& descrs = desc->descriptors;
```

Comment:
Maybe we can use `auto` in all these fields instead of defining `DescriptorMap`

```suggestion
  const auto& descrs = desc->descriptors;
```

Or just move the `using DescriptorMap` that is inside `OSTypeDescriptor` to the `psd` namespace (as I'm seeing this `using` defined in several parts).